api/src/main/java/org/openmrs/module/attachments/obs/AbstractAttachmentHandler.java

```diff
@@ -160,6 +164,7 @@ final public Obs getObs(Obs obs, String view) {
 		
 		ValueComplex valueComplex = new ValueComplex(obs.getValueComplex());
 		if (valueComplex.isOwnImplementation() == false) { // not our implementation
+			view = viewHelper.getView(obs, view);
```

Comment:
If you do this there should not be any need to introduce `DefaultAttachmentHandler2_0`, and the changes made as part of ATT-33 in the controllers should also not be needed.

Reply:
@mks-d The reason I am intoducing this DefaultAttachmentHandler2_0 is because the DefaultAttachmentHandler class sets BinaryDataHandler class of openMRSVersion-1.10.5 which doesnt have the method of supportsview() which was introduced in openMRSVersion-1.12 https://github.com/openmrs/openmrs-core/blob/8b870c34732a9c5ba4f312877eab1f8222b5ad48/api/src/main/java/org/openmrs/obs/ComplexObsHandler.java#L93 thus I am introducing DefaultAttachmentHandler2_0 which relies on this DefaultAttachmentHandler2_0 is because the DefaultAttachmentHandler class sets BinaryDataHandler class of openMRSVersion-2.0.0 otherwise I get your point of not needing the changes in the controller.

Reply:
Did you see my comment [here](https://issues.openmrs.org/browse/ATT-33?focusedCommentId=263116&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-263116)?